portfolio/src/main/webapp/script.js

```diff
@@ -5,18 +5,57 @@ function getRandomQuote() {
     })
 }
 
+let currPage = 0;
+let numCommentsOnPage;
+
 /** Fetch comments from server and add to DOM */
 function getComments() {
     maxNumComments = document.getElementById("num-comments").value;
-    fetch('/data?num=' + maxNumComments).then(response => response.json()).then((comments) => {
+    let prevBtn = document.getElementById("prevBtn")
+
+    handleFirstPage();
+
+    fetch('/data?num=' + maxNumComments + "&page=" + currPage).then(response => response.json()).then((comments) => {
         const commentElement = document.getElementById('video-comments-container');
         commentElement.innerHTML = '';
-        for (i = 0; i < comments.length; i++) {
+        numCommentsOnPage = comments.length;
+
+        handleLastPage();
+
+        for (i = 0; i < numCommentsOnPage; i++) {
             commentElement.appendChild(createCommentElement(comments[i]));
         }
     });
 }
 
+/** If on first page of comments, disable previous button*/
+function handleFirstPage() {
+    if (currPage === 0) {
+        prevBtn.disabled = true;
+    } else {
+        prevBtn.disabled = false;
+    }
+}
+
+/** Handle scenario if on last page of comments*/
+function handleLastPage() {
+    //Disable next button if on last page
+    if (numCommentsOnPage < 5) {
```

Comment:
Am I right in assuming the intent here is to see if there are too few comments on the page such that further pagination on the next button would not result in any new comments? If so, does that need to be dependent on the magic number "5"?

Reply:
Yes that was the intent. I have now changed it so that it takes in a variable (storing the number of comments the user wants to display per page)